### DISCUSSION TITLE
在 WSL2 中使用 Clash for Windows 代理连接 - East Monster 个人博客

### DISCUSSION THREAD

**Discussion by giscus** [EastMonster/eastmonster.github.io#9] [Announcements]
# 在 WSL2 中使用 Clash for Windows 代理连接 - East Monster 个人博客

让 WSL2 下的环境配置更加顺畅

https://eastmonster.github.io/2022/10/05/clash-config-in-wsl/

**Comment by zey9991** [comment]
Brilliant! Problem solved! The instructions are much better than gpt's and other blogs'.

**Comment by TaivasJumala** [comment]
现在更轻松的方法是在 `C:\Users\<your_name>\.wslconfig` 中添加：

```toml
[wsl2]
networkingMode = mirrored
```

**Reply by EastMonster** [reply]
感谢反馈！之前有试过这个方法，但是没成功。现在已找到问题所在，内容已更新。

**Reply by TaivasJumala** [reply]
什么都不改的话，直接开 TUN 模式也行。

**Comment by wsqstar** [comment]
好棒! 但是我也是按照教程来的 ,最后 echo $http_proxy
http://192.168.28.1:10808 $ wget www.google.com
--2024-01-18 15:27:21--  http://www.google.com/
Connecting to 192.168.28.1:10808... failed: Connection refused.

**Reply by EastMonster** [reply]
我这里 `echo $http_proxy` 的结果是 `127.0.0.1:[端口]`. 请问是按照文章顶部的方法还是旧方法设置的？

**Reply by HangVapour** [reply]
我也是这样的，用方法一配置网络镜像，echo $http_proxy 得到 http://主机ip:7890，wget www.google.com 会显示 Connecting to http://主机ip:7890... failed: Connection timed out. 
并且blog的图4，显示楼主 echo $http_proxy 后得到的也是 http://主机ip:端口号，wsl的本地回环地址不指向主机IP

**Reply by HangVapour** [reply]
解决了。source ~/.bashrc 后重启一下，echo $http_proxy
http://127.0.0.1:7890，wget就能访问google。所以配置 .wslconfig 的networkmode= mirrored 之后wsl的本地回环地址就等同主机的本地回环地址

**Reply by Hydra673** [reply]
请问我这里重启之后wsl中输入echo $http_proxy也还是http://172.23.224.1:7897，并不是http://127.0.0.1，wegt google时也链接失败，PS: 
已尝试过开启TURN和关闭DNS、开启镜像，请问是什么问题呢？

**Reply by zhulin4kai** [reply]
@Hydra673 直接把http_proxy写成127.0.0.1就行了，把上面那行host_ip删掉

**Comment by xxxbrem** [comment]
Ubuntu 20.04，wsl 2.0.9.0，使用mirrored方法成功

**Comment by Russellwhatever** [comment]
请问最后最后一步验证成功，输出index.html失败，报错权限不够，请求被拒绝，应该怎么解决啊，我尝试直接chmod 755 index.html失败了

**Reply by EastMonster** [reply]
我没有遇到这个问题，所以帮不了你🙇
`wget` 能把 Google 的网页下载下来就算是成功了，输出什么的无所谓的😂可以自行查找一下解决方法

**Comment by xxxbrem** [comment]
我是在root用户下操作的。wget不能下载的话应该是网络还是没连上吧。

piaoyao ***@***.***> 于2024年1月20日周六 22:55写道：

**Comment by Mingzefei** [comment]
好像必须win11才能使用镜像网络模式？

**Reply by EastMonster** [reply]
~~是的，如果是 Win10 的话就不能用这个方法了~~，之后我会在文章里加上注意事项

**Reply by WaterQun** [reply]
networkingMode=mirrored好像只需要Windows 版本 22H2 或更高版本？Win10 22H2安装WSL2的应该也可以
https://learn.microsoft.com/zh-cn/windows/wsl/wsl-config#main-wsl-settings
Win11最新加入了autoProxy值，配置autoProxy=true可以让WSL直接套用Windows的代理设置，更方便了。
另外，看介绍，autoProxy值是仅在 Windows 11 中可用了。

**Reply by EastMonster** [reply]
感谢反馈！官方文档里那个 "Windows 版本 22H2" 的链接指向的是 Win11 的更新日志，我就想当然了。我也确实搜到了有[在 Win10 上设置镜像网络的文章](https://juejin.cn/post/7323767520167411724)，之后会对文章进行修改。
另外单独设置 `autoProxy` 好像是没用的🤔设置镜像网络就已经把 Windows 的代理应用到 WSL 里面了，暂时不太清楚 `autoProxy` 的用途🙇

**Reply by ybybwdwd** [reply]
win10 22h2(19045.4780)，实测开不了镜像，微软的文档里也明确写了 Windows 11 版本 22H2

**Reply by EastMonster** [reply]
感谢反馈，这下得改回去了。[一月份的时候的文档](https://web.archive.org/web/20240122053636/https://learn.microsoft.com/zh-cn/windows/wsl/wsl-config#main-wsl-settings)写的是 "Windows 版本 22H2"，有点误导。

**Comment by SeeseeYep** [comment]
我在知乎上看见一个类似楼主的贴子，楼主方法二编辑 .bashrc 的时候，两个$host_ip要改成${host_ip}知乎，才能正常使用。另外我的wsl版本是2.0.14

**Reply by EastMonster** [reply]
按照 bash 的语法规则，不加花括号也应该是正确的，`host_ip`后面跟的是冒号而不是字母/数字/下划线，可能当时是别的地方出了问题？🤔

**Comment by Jungle430** [comment]
我按照方法一配置的，用的wsl2和ubuntu22.04，发现`.wslconfig`文件如果里面就是这样
```toml
[wsl2]
networkingMode=mirrored
```
的话，在wsl中，wget命令是好用的，但是DNS解析会失效，所以ping的命令会失败（国内国外的网站都会失败）
还是需要显式指出启用DNS通道
修改配置文件`.wslconfig`为
```toml
[wsl2]
networkingMode=mirrored
dnsTunneling=true
```
这样测试下来的情况是，wget走主机代理，国内国外都能通，ping只能通国内的网站，google还是没法ping通

**Reply by jiaoting** [reply]
因为ping走的是imcp协议，一般翻墙只走tcp协议

**Comment by LewisLiu819** [comment]
请问在使用方法一时出现
`wsl: Mirrored networking mode is not supported, falling back to NAT networking`
该怎么解决？

**Reply by EastMonster** [reply]
无法使用镜像网络这个问题我没法帮到你🙇，仅在连代理这个话题上，你还可以试试方法二。
你可以到 [WSL 的官方仓库](https://github.com/microsoft/WSL/issues) 下面提 issue 试试看

**Comment by momochniu** [comment]
可以先检查一下是否能ping主机。
>PS C:\Users\Administrator> ipconfig
```
以太网适配器 vEthernet (WSL):

   连接特定的 DNS 后缀 . . . . . . . :
   本地链接 IPv6 地址. . . . . . . . : fe80::6b6d:fe24:b809:f259%31
   IPv4 地址 . . . . . . . . . . . . : 172.18.48.1
   子网掩码  . . . . . . . . . . . . : 255.255.240.0
   默认网关. . . . . . . . . . . . . :
```
如果ping不通上面的IPv4地址，powershell中使用
```
New-NetFirewallRule -DisplayName "WSL" -Direction Inbound  -InterfaceAlias "vEthernet (WSL)"  -Action Allow
```
设置一下防火墙规则。

**Reply by Smityz** [reply]
最好使用 telnet <ip> <port> 测试一下端口连通性，如果不连通的话，在防火墙里允许一下 clash 端口

**Comment by fttz** [comment]
太强了！

**Comment by EeureKaaa** [comment]
thanks!

**Comment by duanluan** [comment]
用`export host_ip=$(awk '/nameserver/ {print $2; exit}' /etc/resolv.conf)`避免有多个 IP

**Reply by EastMonster** [reply]
感谢反馈，文章已更新。测试的过程中我发现用[官方文档](https://learn.microsoft.com/zh-cn/windows/wsl/networking#accessing-windows-networking-apps-from-linux-host-ip)提及的获取 IP 的方式会更方便一些，免去了方法二检查 DNS 隧道开启状态的步骤，所以改成了这个。至于这个方法会不会出现多个 IP 的话，暂时忽略，我也不太清楚什么情况会有多个 IP 🤔

**Comment by CN-CODEGOD** [comment]
wsl: experimental.useWindowsDnsCache:C:\Users\Administrator\.wslconfig 中的键“13”未知
wsl: Hyper-V 防火墙不受支持
wsl: 不支持镜像网络模式： Windows 版本 19045。3803 没有所需的功能。
正在回退到 NAT 网络。
wsl: DNS 隧道不受支持
administrator@laptop-latitude:/mnt/o/OneDrive$ wget www.google.com
--2025-09-27 15:53:23--  http://www.google.com/
Connecting to 172.25.240.1:7890...


Windows 10 19045 没有镜像功能 所以用了第二种方法结果连接不上
